src/TimerManager.ts

```diff
@@ -34,6 +49,30 @@ class TimerManager {
         return timeout;
     }
 
+    generateSetInterval(planStep:PlanStep) {
+        return (callback: (...args: any[]) => void, delay: number, ...args: any[]): Interval => {
+            // TODO: wrap the callback to track last run time, run counter, and other stats
+            const interval: Interval = setInterval(callback, delay, ...args);
+            this.intervals.add(interval);
+            this.jsonPointerByInterval.set(interval, planStep.jsonPtr);
+            this.wrappedIntervals.add({interval: interval, jsonPointerStr: planStep.jsonPtr});
+            return interval;
+        }
+    }
+
+    generateClearInterval(planStep:PlanStep) {
+        return async (interval: Interval): Promise<void> => {
+            clearInterval(interval);
+            const jsonPointerStr: string | undefined = this.jsonPointerByInterval.get(interval);
+            if (jsonPointerStr != undefined) {
+                // jp.set(this.tp.output, jsonPointerStr, "--deleted-interval--");
+                // jp.set(this.tp.templateMeta, jsonPointerStr, "--deleted-interval--");
+                await this.tp.setData(jsonPointerStr, "--deleted-interval--", "forceSetInternal");
```

Comment:
let's discuss if this should await `setData` or simply "launch" it and move on. Probably it should be await, as you have it.